File: chapter3/java/designpattern/src/main/java/com/designpattern/PizzaBuilder.java
package com.designpattern;

/**
 * PizzaBuilder interface defines common build method
 *
 * @Author: Bridget
 */
public interface PizzaBuilder {
    PizzaBuilder addSauce(String sauce);

    PizzaBuilder addTopping(String topping);

    PizzaBuilder buildSize(String size);

    PizzaBuilder createPizza();

    Pizza getPizza();
}

File: chapter3/typescript/PizzaBuilder.ts
import { Pizza } from "./Pizza";

/**
 * PizzaBuilder interface defines common build method
 * 
 * @Author: Bridget
 */
 export interface PizzaBuilder {

    addSauce(sauce: string): PizzaBuilder;
    addTopping(topping: string): PizzaBuilder;
    buildSize(size: string): PizzaBuilder;
    createPizza(): PizzaBuilder;
    getPizza(): Pizza;

}
